### PR TITLE
New airframe for ThunderFly Auto-G2 autogyro

### PR DESCRIPTION
**Pull request by roman-dvorak**
We would like to merge our autogyro ThunderFly AutoG2 airframe and we also need to create a new group of airframes for autogyros.

This airframe supposes an autogyro with a rotor controlled in two axes (pitch and roll), a rudder and an elevator. Optionally it can be equipped with a release device for a tow rope for take-off. An airspeed sensor is recommended.

Our test drone is based on the Durafly Auto-G2 with a replaced 3D printed rotor-head (exchanged for the rotor-head with two axes of freedom), an added autopilot, an airspeed sensor and optionally a release device and 3D printed rotor blades. All these 3D printed mechanical parts can be found in [ThunderFly Auto-G2](https://github.com/ThunderFly-aerospace/Auto-G2/) repository.

In the near future, we are planning to assemble an autogyro training model made almost entirely of 3D printed parts.

Video records of test flights: 
https://youtu.be/YhXXSWz5wWs
https://youtu.be/taJ1aLBtanQ

### CODE REVIEW

**Comment by roman-dvorak** [comment]
This PR contains a mapping for the autogyro group icon. It is related to the next pull-requests in corresponding repositories.

QGC: https://github.com/mavlink/qgroundcontrol/pull/8229
Dev documentation: prepared 
User documentation: prepared

**Comment by dagar** [comment]
Very cool! It's great to see new uses like this.

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+#
+# @name TF:AutoG2
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]:
```suggestion
# @name  ThunderFly:AutoG2
```
?

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#
+# @output MAIN1 rotor_head_L
+# @output MAIN2 rotor_head_R
+# @output MAIN3 elevator
+# @output MAIN4 rudder
+# @output MAIN5 rudder (second, optional)
+# @output MAIN6 throttle
+# @output MAIN7 wheel
+#
+# @output AUX1 prerotator (optional)
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Based on the current config, there is no AUX mixer loaded, or do I miss something?

**Comment by Jaeyoung-Lim** [comment]
@roman-dvorak Very cool! Is there any flight log available?

**Comment by julianoes** [comment]
FYI @JonasVautherin

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 (hunk 1 above)
Comment (roman-dvorak) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Yes, this change is acceptable from us.

**Comment by roman-dvorak** [comment]
> @roman-dvorak Very cool! Are there any flight log available?
Yes, there is one our flight log.
https://review.px4.io/plot_app?log=7efc4214-6d19-4146-95a9-6793eb36c002

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 (hunk 2 above)
Comment (roman-dvorak) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Aux pass mixer was added in new commit: https://github.com/PX4/Firmware/pull/14015/commits/dc357dc4147f4b12b62fd38b027ec31cd6c23ca9

Comment (bkueng) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Then I'd rename this to `feed-through of RC AUX1 channel`

Comment (roman-dvorak) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Ok, description is chaneged

https://github.com/PX4/Firmware/blob/76b3ff04598f5a4a4a4e1487a93386d718cf4643/ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2#L16-L17

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 (hunk 1 above)
Comment (roman-dvorak) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Renamed

https://github.com/PX4/Firmware/blob/76b3ff04598f5a4a4a4e1487a93386d718cf4643/ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2#L3

File: ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2 [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+if [ $AUTOCNF = yes ]
+then
+	param set BAT_CAPACITY 2500.0
+	param set BAT_N_CELLS 3
+
+	param set PWM_AUX_RATE 50
+	param set PWM_RATE 50
+
+    param set SENS_BOARD_ROT 8
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: Whitespace: can you use tabs?

Comment (roman-dvorak) [reply ROMFS/px4fmu_common/init.d/airframes/17002_TF-AutoG2]: repaired.
Thanks for the warning. I didn't know about this convention.

**Pull request merged**